### DISCUSSION TITLE
RingBufferStream over underflow control .

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#811] [Q&A] [answered]
i want to use ring buffer stream ,but cant control the actual position how much it its filled . 
i used available but it gives me always 1024 which is default value , is this ok ? ? how to know its is over flow or under flow ?  
  

RingBufferStream rngbuffer;
.
int status = rngbuffer.available();-------------------> always 1024

how to do this ?

**Comment by pschatzmann** [comment, marked as answer]
I think you talk nonsense: available() will provide the number of entries that are available in the buffer and availableForWrite() provides the number of entries that can still be written: So your buffer is full!

There is no surprise here as this is valid for all Arduino Streams like e.g. Serial
and it continues to be full as long as you don't start to read from it!

You can define the size in the constructor or call resize().
